pkg/controller/realm/index.go

```diff
@@ -48,8 +49,20 @@ func (c *Controller) HandleIndex() http.Handler {
 		if len(userRealms) == 1 {
 			realm := userRealms[0]
 
+			if controller.RealmIDFromSession(session) == realm.ID {
+				http.Redirect(w, r, "/home", http.StatusFound)
+				return
+			}
+
 			controller.StoreSessionRealm(session, realm)
 			flash.Alert("Logged into verification system for '%s'", realm.Name)
+
+			if realm.MFAMode == database.MFAOptionalPrompt {
+				if factors := controller.FactorCountFromSession(session); factors == 0 {
+					http.Redirect(w, r, "/login/registerphone", http.StatusSeeOther)
```

Comment:
Need `return`

Reply:
It'd also be nice to move this logic into the controller since I've seen it at least twice (e.g. `controller.RedirectToMFA`)